private/buf/bufstudioagent/bufstudioagent_test.go

```diff
@@ -193,14 +194,31 @@ func testPlainPostHandlerErrors(t *testing.T, upstreamServer *httptest.Server) {
 	})
 
 	t.Run("invalid_upstream", func(t *testing.T) {
-		listener, err := net.Listen("tcp", "127.0.0.1:")
+		tcpAddr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:")
 		require.NoError(t, err)
+		listener, err := net.ListenTCP("tcp", tcpAddr)
+		require.NoError(t, err)
+		done := make(chan struct{})
+		t.Cleanup(func() {
+			done <- struct{}{}
+		})
 		go func() {
-			conn, err := listener.Accept()
-			require.NoError(t, err)
-			require.NoError(t, conn.Close())
+			for {
+				select {
+				case <-done:
+					listener.Close()
```

Comment:
Why the channel? It's equivalent to call `listener.Close()` in the Cleanup and you already must handle the error on accept.

Reply:
If I call `listener.Close()` from the parent goroutine, the listener wouldn't know it's closed (plz correct me if I'm wrong) and then would just keep calling `Accept` even though it's already closed.

Reply:
Ah the continue would need to do error handling. Maybe this is fine.